nexus/src/db/datastore.rs

```diff
@@ -379,6 +398,26 @@ impl DataStore {
                     .returning(Region::as_returning())
                     .get_results(conn)?;
 
+                // Update the tallied sizes in the source datasets containing
+                // those regions.
+                let region_size = i64::from(params.block_size())
+                    * params.blocks_per_extent()
+                    * params.extent_count();
+                for dataset in source_datasets.iter_mut() {
+                    dataset.size_used =
+                        dataset.size_used.map(|v| v + region_size);
+                }
+
+                let dataset_ids: Vec<Uuid> =
+                    source_datasets.iter().map(|ds| ds.id()).collect();
+                diesel::update(dataset_dsl::dataset)
+                    .filter(dataset_dsl::id.eq_any(dataset_ids))
+                    .set(
+                        dataset_dsl::size_used
```

Comment:
How so?

- `region_size` does not risk being modified; it's calculated based on the size of the request
- The SQL operation here is basically:

```sql
UPDATE "dataset"
SET "size_used" = ("dataset"."size_used" + $1)
WHERE ("dataset"."id" = ANY($2))
```

Since this is an individual statement, I figured it fell under the [transaction rules](https://www.cockroachlabs.com/docs/v21.2/transactions#individual-statements) for CRDB, meaning that it is atomic.

So, basically, even if there are concurrent modifications changing "size_used", this operation will atomically update the value "up by size_used", or "down by size_used" in the de-allocation path below.

Reply:
Got it -- I misunderstood what SQL would be generated for that expression.  That looks right.  Thanks for clarifying!